rules/expiring-todo-comments.js

```diff
@@ -0,0 +1,285 @@
+'use strict';
+const readPkg = require('read-pkg');
+const semver = require('semver');
+const baseRule = require('eslint/lib/rules/no-warning-comments');
+const getDocsUrl = require('./utils/get-docs-url');
+
+const pkg = readPkg.sync();
+
+const pkgDependencies = {...pkg.dependencies, ...pkg.devDependencies};
+
+const TODO_RE = /[TODO|FIXME][\s\S]*\[([^}]+)\]/i;
+const DEPENDENCY_INCLUSION_RE = /^[+|-]\s*@?[\S+]\/?\S+/;
+const DEPENDENCY_VERSION_RE = /^(@?[\S+]\/?\S+)@(>|>=)([\d]+(\.\d+){0,2})/;
+const PKG_VERSION_RE = /^(>|>=)([\d]+(\.\d+){0,2})\s*$/;
+const ISO8601 = /(\d{4})-(\d{2})-(\d{2})/;
+
+const create = context => {
+	const sourceCode = context.getSourceCode();
+	const comments = sourceCode.getAllComments();
+	const unnusedComments = comments
+		.filter(token => token.type !== 'Shebang')
+		.filter(processComment);
+
+	// This is highly dependable on eslint's no-warning-comments implementation.
+	// What I do is patch the parts I know the rule will use, the getAllComments.
+	// Since I have priority, I leave only the comments that I didn't use to it.
+	const fakeContext = {...context, getSourceCode() {
```

Comment:
@sindresorhus I'd love your comment on this approach.

Unlike typescript-eslint [approach](https://github.com/typescript-eslint/typescript-eslint/blob/4edf0d71ef0a8a3370165e804dd693033d6207bc/packages/eslint-plugin/src/rules/no-unused-vars.ts#L42) where they could extend the visitor functions, I had not such ideal scenario as the default rule read nothing but the context.